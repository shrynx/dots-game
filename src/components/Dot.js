// @flow
import styled from 'styled-components';

type Props = {
  radius: number,
  backgroundColor: string,
  left: number,
  top: number,
};

const Dot = styled.button.attrs({
  style: ({ radius, backgroundColor, top, left }: Props) => ({
    height: radius,
    width: radius,
    backgroundColor,
    top,
    left,
  }),
})`
  border-radius: 50%;
  opacity: 0.8;
  position: absolute;
  &:hover {
    cursor: cell;
  }
`;

export default Dot;
