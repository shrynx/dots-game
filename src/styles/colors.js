// @flow
export const YELLOW = '#ffc761';
export const RED = '#e64455';
export const CYAN = '#00b4ce';
export const MAGENTA = '#ba55d3';
export const GREEN = '#28a26a';
export const PURPLE = '#4400cc';
export const GREY = '#828282';
export const BACKGROUND_COLOR = '#222232';
