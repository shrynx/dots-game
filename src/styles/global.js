// @flow
import { injectGlobal } from 'styled-components';

export default injectGlobal`
  body {
    padding: 0;
    margin: 0;
    font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
      'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue', sans-serif;
    -webkit-font-smoothing: antialiased;
    -moz-osx-font-smoothing: grayscale;
  }

  button {
    border: none;
    background: none;
    padding: 0;
    -webkit-appearance: none;
  }

  button:focus {
    outline: none;
  }
`;
