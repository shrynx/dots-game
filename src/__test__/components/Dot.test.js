import React from 'react';
import { render } from 'react-testing-library';
import Dot from '../../components/Dot';
import { RED } from '../../styles/colors';

it('Dot component should render correctly', () => {
  const { container } = render(<Dot radius={5} backgroundColor={RED} left={100} top={100} />);
  expect(container).toMatchSnapshot();
});
