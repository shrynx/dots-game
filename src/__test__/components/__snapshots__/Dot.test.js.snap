// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dot component should render correctly 1`] = `
<div>
  <button
    style="height: 5px; width: 5px; background-color: rgb(230, 68, 85); top: 100px; left: 100px;"
    class="sc-bdVaJa dyawHZ"
    radius="5"
  />
</div>
`;
